### DISCUSSION TITLE
Blazor Bootstrap v3.0.0-preview.2 released!!!

### DISCUSSION THREAD

**Discussion by gvreddy04** [vikramlearning/blazorbootstrap#784] [Announcements]
## What's new

- Polar Area Chart component
- Image component

## What's changed

- Grid component
  - Details View
  - Nested Grid
  - Templates: New GridLoadingTemplate and GridEmptyDataTemplate
  - New GridContainerClass and GridContainerStyle parameters
  - Highlight row when selected and customize the row color and background color
  - All the demos updated

- Sidebar component
  - New Width and WidthUnit parameters

- Sidebar2 component
  - New Width and WidthUnit parameters

- Demos & Docs updated.

## Links

Demos - Blazor Server: https://demos.blazorbootstrap.com/
Demos - Blazor WebAssembly: https://demos.getblazorbootstrap.com/
Docs: https://docs.blazorbootstrap.com/
GitHub: https://github.com/vikramlearning/blazorbootstrap
NuGet: https://www.nuget.org/packages/blazor.bootstrap

#bootstrap #blazorbootstrap #charts #grid #image #polarareachart #sidebar #sidebar2

**Comment by prizmasoftware** [comment]
Thanks for the code addition in terms of the ability to insert a subgrid into the grid control.
I've looked at examples, but I haven't been able to find a way to associate the ID of the rows from base grid with the list of subgrids in terms of the base_grid.rows.id=subgrid.rows.id_grid relation, when using two different table (lists), and subgrid content depends of main grid choice...
For the sub grid, I provide the data via dataprovider, but I cannot transfer the information about the grid.rows.ID (context.ID).
If it's not a problem, I would like to ask for help.
Thank you.

**Reply by prizmasoftware** [reply]
I solved it, maybe it will help someone:

GridDetailView TItem="NormativeInfo" Context="normative">
Grid
  .......
  DataProvider="(request) => GetData(request, normative.Id)">
/Grid

**Comment by prizmasoftware** [comment]
There is another problem with sorting and displaying subgrids...
E.g:
Grid 1
----- grid 1.1
----- grid 1.2
---------- subgrid 1.2.1
---------- subgrid 1.2.2
----- grid 1.3
----- grid 1.4

this is how the list of grids and subgrids was formed during initialization, and this is how it looks after clicking on grid 1.2
When I sort the grid in descending order the display looks like this:

----- grid 1.4
----- grid 1.3
---------- subgrid 1.2.1
---------- subgrid 1.2.2
----- grid 1.2
----- grid 1.1

how to refresh after sorting?
